pkg/virt-handler/node-labeller/node_labeller.go

```diff
@@ -0,0 +1,183 @@
+package nodelabeller
+
+import (
+	goerror "errors"
+	"fmt"
+	"strconv"
+	"strings"
+
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/client-go/tools/cache"
+
+	"kubevirt.io/client-go/kubecli"
+	"kubevirt.io/client-go/log"
+	device_manager "kubevirt.io/kubevirt/pkg/virt-handler/device-manager"
+)
+
+const (
+	labelNamespace    = "feature.node.kubernetes.io"
+	labellerNamespace = "node-labeller"
+)
+
+var nodeLabellerVolumePath = "/var/lib/kubevirt-node-labeller"
+
+//NodeLabeller struct holds informations needed to run node-labeller
+type NodeLabeller struct {
+	kvmController     *device_manager.DeviceController
+	configMapInformer cache.SharedIndexInformer
+	clientset         kubecli.KubevirtClient
+	host              string
+	namespace         string
+}
+
+func NewNodeLabeller(kvmController *device_manager.DeviceController, configMapInformer cache.SharedIndexInformer, clientset kubecli.KubevirtClient, host, namespace string) *NodeLabeller {
+	return &NodeLabeller{
+		kvmController:     kvmController,
+		configMapInformer: configMapInformer,
+		clientset:         clientset,
+
+		host:      host,
+		namespace: namespace,
+	}
+}
+
+//Run runs node-labeller
+func (n *NodeLabeller) Run(stop chan struct{}) {
+	logger := log.DefaultLogger()
+	logger.Infof("node-labeller is running")
+
+	n.configMapInformer.AddEventHandler(cache.ResourceEventHandlerFuncs{
+		AddFunc: func(obj interface{}) {
+			err := n.run()
+			if err != nil {
+				logger.Infof("something happened during labelling node %s at config map creatinon: %v\n", n.host, err)
+			}
+		},
+		DeleteFunc: func(obj interface{}) {
+			err := n.run()
+			if err != nil {
+				logger.Infof("something happened during labelling node %s at config map deletion: %v\n", n.host, err)
+			}
+		},
+		UpdateFunc: func(old, new interface{}) {
+			err := n.run()
+			if err != nil {
+				logger.Infof("something happened during labelling node %s at config map update: %v\n", n.host, err)
+			}
+		},
+	})
+
+	go n.configMapInformer.Run(stop)
```

Comment:
i think we need a node informer here as well. Isn't the goal to ensure the nodes always have the correct labels?

Reply:
you mean it e.g. like if user deletes all labels?
Because right now we just need to relabel only when node-labeller config map change.

Reply:
right, so since we're applying the label to the node, it seems like we should be watching the nodes to ensure the labels are set and remain set.